spec/realtime/message.test.js

```diff
@@ -70,6 +70,50 @@ define(['ably', 'shared_helper'], function(Ably, helper) {
 		}
 	};
 
+	/*
+	 * Test that a message is not sent back to the same realtime client
+	 * when echoMessages is false (RTC1a and RTL7f)
+	 *
+	 * Test that a message is sent back to the same realtime client
+	 * when echoMessages is true (RTC1a and RTL7f)
+	 */
+	exports.publishEcho = function(test) {
+		test.expect(3);
+
+		// set up two realtimes
+		var rt1 = helper.AblyRealtime({ echoMessages: false }),
+			rt2 = helper.AblyRealtime({ echoMessages: true }),
+
+			rt1Channel = rt1.channels.get('publishecho'),
+			rt2Channel = rt2.channels.get('publishecho'),
+
+			testMsg1 = 'Hello',
+			testMsg2 = 'World!';
+
+		monitorConnection(test, rt1);
+		monitorConnection(test, rt2);
+
+		// We expect to see testMsg2 on rt1 and testMsg1 on both rt1 and rt2
+		var expectedMessages = 3;
+
+		rt1Channel.subscribe('event0', function(msg) {
+			test.equal(msg.data, testMsg2, 'Received testMsg2 via rt1 instance');
+			if (!--expectedMessages)
+				closeAndFinish(test, [rt1, rt2]);
+		});
+
+		rt2Channel.subscribe('event0', function(msg) {
+			test.ok( msg.data == testMsg1 || msg.data == testMsg2, 'Received testMsg1 or 2 via rt2 instance' );
+			if (!--expectedMessages)
+				closeAndFinish(test, [rt1, rt2]);
+		});
+
+		// publish testMsg1 on rt1 and testMsg2 on rt2
+		rt1Channel.publish('event0', testMsg1, function() {
```

Comment:
This test is too brittle because rt2Channel.subscribe may attach and subscribe after rt1Channel has published it's message and thus never receive it.  You should wait until the attaches are finished or approach this differently.